systems/console-bff/subscriptions/resolvers/resolver.ts

```diff
@@ -335,271 +335,329 @@ class SubscriptionsResolvers {
+    const nodeResults: MetricsStateRes = { metrics: [] };
+    if (metricKeys.length > 0) {
+      const metricPromises = metricKeys.map(async key => {
+        const res = await getNodeMetricRange(baseURL, key, { ...data, nodeId });
+        if (Array.isArray(res.metrics)) {
+          res.metrics.forEach(metric => {
+            nodeResults.metrics.push(processMetricResult(metric));
+          });
+        }
+      });
+      await Promise.all(metricPromises);
+    }
+    return nodeResults.metrics.map(res => ({
+      msg: res.msg,
+      type: res.type,
+      siteId: siteId || "",
+      nodeId: nodeId || "",
+      success: res.success,
+      value: formatKPIValue(res.type, res.value),
+    }));
+  }
 
-            for (const key of metricKeys) {
-              metrics.metrics.push({
-                msg: "Error",
-                type: key,
-                siteId: siteId || "",
-                nodeId: nodeId || "",
-                success: false,
-                value: 0,
-              });
-            }
+  private setupSiteStateWSWorkers(
+    wsUrl: string,
+    metricKeys: string[],
+    data: GetMetricsSiteStatInput,
+    siteIds: string[],
+    nodeIds: string[] | undefined,
+    baseTopic: string
+  ) {
+    const handleSiteWSMessage = (
+      wsData: any,
+      topic: string,
+      siteId: string,
+      nodeId: string = ""
+    ) => {
+      if (!wsData.isError) {
+        try {
+          const res = JSON.parse(wsData.data);
+          if (res?.data?.result && res.data.result.length > 0) {
+            res.data.result.forEach((result: any) => {
+              if (
+                result &&
+                result.metric &&
+                result.value &&
+                result.value.length > 0
+              ) {
+                if (!nodeId) {
+                  const resultSiteId =
+                    result.metric.site || result.metric.instance || siteId;
+
+                  pubSub.publish(topic, {
+                    success: true,
+                    msg: "success",
+                    type: res.Name,
+                    siteId: resultSiteId,
+                    nodeId: "",
+                    value: [
+                      Math.floor(result.value[0]) * 1000,
+                      formatKPIValue(res.Name, result.value[1]),
+                    ],
+                  });
+                } else {
+                  const resultNodeId =
+                    result.metric.node || result.metric.nodeid || nodeId;
+
+                  pubSub.publish(topic, {
+                    success: true,
+                    msg: "success",
+                    type: res.Name,
+                    siteId: siteId,
+                    nodeId: resultNodeId,
+                    value: [
+                      Math.floor(result.value[0]) * 1000,
+                      formatKPIValue(res.Name, result.value[1]),
+                    ],
+                  });
+                }
+              }
+            });
```

Comment:
Includes so unnecessary if else. I think you can use `handleWebSocketMessage` method which is doing the same stuff i believe.
ref: https://github.com/ukama/ukama/blob/be9dde24907bd49be6b840eed83b237cc1cea2fc/systems/console-bff/subscriptions/resolvers/resolver.ts#L76-L106

Reply:
addressed